src/controllers/friendRequestController.js

```diff
@@ -0,0 +1,159 @@
+const { client } = require("../config/configDB");
+const { removeElementFromSet } = require("../utils/cache");
+const friendRequestLookup = require("../utils/friendRequestLookup");
+const friend_requests_cache = new Set()
+const tableContainsLink = require("../utils/tabelContainsLink");
+const friendship_cache = new Set()
+
+
+const getAllFriendRequestObjectsByUserId = (req, res) => {
+    client.query(`
+    SELECT * FROM friend_requests
+    WHERE user1_id = $1 or user2_id = $1
+    `, [req.user.id], (err, results) => {
+        if (err) {
+            return res.status(500).json({ status: 500, message: "unknown error occurred while fetching all the requests" })
+        }
+        else if (results.rows[0].length !== 0) {
+            return res.status(200).json(results)
+        }
+        else {
+            return res.status(400).json({ status: 400, message: "No friend requests were found" });
+        }
+    })
+}
+
+async function createFriendRequestObject(req, res, next) {
+    try {
+        const { user1_id, user2_id } = req.body
+        if (req.user.id !== user1_id && req.user.id !== user2_id) {
+            return res.status(404).json({ status: 404, message: "User is not allowed to perform this operation" })
+        }
+        const requestAlreadyExists = await tableContainsLink("friend_requests", "user1_id", "user2_id", user1_id, user2_id, friend_requests_cache)
+        const usersAlreadyFriends = friendship_cache.has(String(user1_id) + '-' + String(user2_id))
+
+        if (usersAlreadyFriends) {
+            res.status(400).json({ status: 400, message: "Invalid request: Users are already friends" })
+        }
+        else if (requestAlreadyExists) {
+            res.status(400).json({ status: 400, message: "Invalid request: The request object already exists, withdraw it, or wait till it is accepted/denied" })
+        }
+        else {
+
+            client.query("INSERT INTO friend_requests (user1_id, user2_id, request_status) VALUES ($1, $2, $3) RETURNING *",
```

Comment:
These fields are not enough to see who the requester is, and who the requestee is.
Without this info we will not be able to show for example, all the requests that user 1 has gotten. etc because we will not know who sent this request.